samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -160,7 +158,7 @@ public void close() throws ZkInterruptedException {
     * @param dataListener describe this
     */
   public void subscribeToJobModelVersionChange(IZkDataListener dataListener) {
-    LOG.info("pid=" + processorId + " subscribing for jm version change at:" + keyBuilder.getJobModelVersionPath());
+    LOG.info(" subscribing for jm version change at:" + keyBuilder.getJobModelVersionPath());
```

Comment:
getJobModelVersion Path returns the path to the znode where the version is stored. The paths are managed by ZkKeyBuilder. 

newJobModelVersion field was not used anywhere in the ZkJobCoordinator. As I explained above, I only refactored as it was not actually used by ZkJobCoordinator.

Let's sync-up in-person if you have further questions.